WordPress/gutenberg
# Global Styles: update metadata and add support for padding

This PR does a couple of things:

- Reduces the amount of metadata we send to the client. Before this PR we sent all the metadata for every context, now the client derives the contexts from existing data. Fixes https://github.com/WordPress/gutenberg/issues/26802

- While we allowed padding to be set at the block level (block attributes) we didn't do the same for the global level (theme.json). This is a bug that happened due to the distributed nature of the metadata that we had. Note that this needs a follow-up PR to add the padding control to the global styles sidebar.

## How to test

Test that it works as before:

- Install and activate a theme that supports theme.json (ex: TwentyTwentyOne blocks).
- Check that generated styles correspond to the theme's theme.json configuration (in front-end, edit-site, and edit-post).

Tests that it padding works properly:

- Install and activate a theme that supports theme.json (ex: TwentyTwentyOne blocks).
- Declare some padding value for the contexts that support it (cover or group).
- Check that the generated styles have the padding property.
